test/Compiler.test.js

```diff
@@ -190,6 +190,25 @@ describe("Compiler", () => {
 		});
 	});
 
+	it("compiles top-level await to es5", done => {
+		compile(
+			"./topLevelAwait1",
+			{
+				target: ["web", "es5"],
+				experiments: {
+					topLevelAwait: true
+				}
+			},
+			(stats, files) => {
+				expect(Object.keys(files)).toEqual(["/main.js"]);
+				const bundle = files["/main.js"];
+				expect(bundle).not.toContain("await");
+				expect(bundle).not.toContain("async");
+				done();
+			}
+		);
+	});
+
```

Comment:
I created a gist that shows what the file currently looks like and below I tried to manually rewrite what could work in an ES5 environment: https://gist.github.com/amannn/998ddbd6f1924c50d21d9450c2aa45ff.

It seems like [`AwaitDependenciesInitFragment`](https://github.com/webpack/webpack/blob/master/lib/async-modules/AwaitDependenciesInitFragment.js) can be used to prepend the call site of a module. But how would you be able to wrap a `.then(…)` around the whole module consumption?

Maybe you have some pointers @evilebottnawi?